third_party/meerkat/Component/mmDiscovery/service_server.cpp

```diff
@@ -110,18 +110,20 @@ VOID CServiceServer::DataRecv(OSAL_Socket_Handle iEventSock,
     std::vector<char*> argv;
     t_HandlePacket(argv, pData + strlen(kServiceRequestScheme));
 
-    char server_address[35] = {'\0',};
-    snprintf(server_address, sizeof(server_address) - 1,
-             "--enable-castanets=%s", pszsource_addr);
-    argv.push_back(server_address);
-
-    // TODO: |server_address_old| should be remove after applying
-    // https://github.com/Samsung/Castanets/pull/75.
-    char server_address_old[35] = {'\0',};
-    snprintf(server_address_old, sizeof(server_address_old) - 1,
-             "--server-address=%s", pszsource_addr);
-    argv.push_back(server_address_old);
-
+    const std::string switches(argv.front());
+    if (!switches.find("--enable-service-offloading")) {
+      char server_address[35] = {'\0',};
+      snprintf(server_address, sizeof(server_address) - 1,
+               "--enable-castanets=%s", pszsource_addr);
+      argv.push_back(server_address);
+
+      // TODO: |server_address_old| should be remove after applying
+      // https://github.com/Samsung/Castanets/pull/75.
+      char server_address_old[35] = {'\0',};
```

Comment:
I don't think |server_address_old| is necessary.

Reply:
I just don't want to change original code.

Reply:
I asked @younghajung and noticed that the added address is needed for renderer offloading on castanets.

Reply:
The address is appended with --enable-castanets.